disnake/welcome_screen.py

```diff
@@ -0,0 +1,223 @@
+    )
+
+    def __init__(
+        self,
+        *,
+        id: int,
+        description: str,
+        emoji: Optional[Union[str, Emoji, PartialEmoji]] = None,
+    ):
+        self.id: int = id
+        self.description: str = description
+        self.emoji: Optional[PartialEmoji] = None
+        if emoji is None:
+            self.emoji = None
+        elif isinstance(emoji, str):
+            self.emoji = PartialEmoji.from_str(emoji)
+        elif isinstance(emoji, PartialEmoji):
+            self.emoji = emoji
+        elif isinstance(emoji, _EmojiTag):
+            self.emoji = emoji._to_partial()
+        else:
+            raise TypeError("emoji must be None, a str, PartialEmoji, or Emoji instance.")
+
+    def __repr__(self):
+        return f"<WelcomeScreenChannel id={self.id!r} emoji={self.emoji!r} description={self.description!r}>"
+
+    @classmethod
+    def _from_data(
+        cls, *, data: WelcomeScreenChannelPayload, state: ConnectionState
+    ) -> WelcomeScreenChannel:
+        emoji_id = _get_as_snowflake(data, "emoji_id")
+        emoji_name = data.get("emoji_name") or None
+        emoji = None
+        if emoji_name:
+            emoji = PartialEmoji.with_state(state, name=emoji_name, id=emoji_id)
+
+        return cls(id=int(data["channel_id"]), description=data["description"], emoji=emoji)
+
+    def to_dict(self) -> WelcomeScreenChannelPayload:
+
+        result = {
+            "channel_id": self.id,
+            "description": self.description,
+        }
+        if self.emoji is not None:
+            if self.emoji.id:
+                result["emoji_id"] = self.emoji.id
+            result["emoji_name"] = self.emoji.name
+
+        return result  # type: ignore
+
+
+class WelcomeScreen:
+    """Represents a Discord welcome screen for a :class:`Guild`.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    description: Optional[:class:`str`]
+        The guild description in the welcome screen.
+    channels: List[:class:`WelcomeScreenChannel`]
+        The welcome screen's channels.
+    """
+
+    __slots__ = (
+        "description",
+        "welcome_channels",
+        "_guild",
+        "_state",
+    )
+
+    def __init__(
+        self,
+        *,
+        data: WelcomeScreenPayload,
+        state: ConnectionState,
+        guild: Union[Guild, PartialInviteGuild],
+    ):
+        self._state = state
+        self._guild = guild
+        self.description: Optional[str] = data.get("description")
+        self.welcome_channels: List[WelcomeScreenChannel] = [
+            WelcomeScreenChannel._from_data(data=channel, state=state)
+            for channel in data["welcome_channels"]
+        ]
+
+    def __repr__(self) -> str:
+        return f"<WelcomeScreen description={self.description!r} welcome_channels={self.welcome_channels!r} enabled={self.enabled!r}>"
+
+    @property
+    def enabled(self) -> bool:
+        """:class:`bool`: Whether the welcome screen is displayed to users.
```

Comment:
Should maybe note that this doesn't take into account whether the community feature is enabled